### PR TITLE
Fix remaining failing CPU linux tests.

### PR DESCRIPTION
**Pull request by pschuh**
test/SILGen/builtins.swift -> Extra assert change hanging around.
utils/PathSanitizingFileCheck -> injection bug.
validation-test/ParseableInterface/verify_all_overlays.py -> Just a fix.

### CODE REVIEW

**Comment by pschuh** [comment]
@swift-ci please test tensorflow linux

File: validation-test/ParseableInterface/verify_all_overlays.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
     else:
         continue
 
-    if module_name == "Swift" or module_name == "SwiftLang":
+    if module_name in ["Swift", "SwiftLang", "DifferentiationUnittest", "Python", "TensorFlow"]:
```
Comment (dan-zheng) [inline validation-test/ParseableInterface/verify_all_overlays.py]: Could you please make this change instead?
```
diff --git a/validation-test/ParseableInterface/verify_all_overlays.py b/validation-test/ParseableInterface/verify_all_overlays.py
index 6fb203fa15..778322b1c9 100755
--- a/validation-test/ParseableInterface/verify_all_overlays.py
+++ b/validation-test/ParseableInterface/verify_all_overlays.py
@@ -14,6 +14,9 @@

 # Expected failures by platform
 # -----------------------------
+# SWIFT_ENABLE_TENSORFLOW
+# FIXME: Re-enable for DifferentiationUnittest, Python, and TensorFlow modules.
+# macosx: DifferentiationUnittest
+# macosx: Python
+# macosx: TensorFlow
 # macosx: XCTest
 # ios: XCTest
 # tvos: XCTest
```

File: test/SILGen/builtins.swift [hunk 2]
```diff
@@ -577,13 +577,6 @@ func castBitPatternFromBridgeObject(_ bo: Builtin.BridgeObject) -> Builtin.Word
   return Builtin.castBitPatternFromBridgeObject(bo)
 }
 
-// SWIFT_ENABLE_TENSORFLOW
```
Comment (dan-zheng) [inline test/SILGen/builtins.swift]: I wonder, is it not possible to easily fix this test?

Comment (dan-zheng) [reply test/SILGen/builtins.swift]: @marcrasi confirmed that this test should be removed. Tests for `poundAssert` exist upstream.

File: utils/PathSanitizingFileCheck [hunk 3]
```diff
@@ -72,7 +72,7 @@ constants.""")
         # We are replacing the Unix path separators in the paths passed as
         # arguments with a broader pattern to also allow forward slashes and
         # double escaped slashes in the result that we are checking. Sigh.
-        stdin = re.sub(re.sub(r'/', slashes_re, pattern), replacement, stdin)
+        stdin = re.sub(re.sub(r'\/', slashes_re, re.escape(pattern)), replacement, stdin)
```
Comment (dan-zheng) [inline utils/PathSanitizingFileCheck]: Could you please upstream this change to `master` and mark with `# SWIFT_ENABLE_TENSORFLOW` in the meantime?

File: validation-test/ParseableInterface/verify_all_overlays.py (hunk 1 above)
Comment (pschuh) [reply validation-test/ParseableInterface/verify_all_overlays.py]: This only seems sufficient for macosx?

Comment (dan-zheng) [reply validation-test/ParseableInterface/verify_all_overlays.py]: Ah, good point. Could you please add `SWIFT_ENABLE_TENSORFLOW` to the current solution instead?

**Review by dan-zheng — APPROVED** [review]

**Pull request merged**